source/particles/particle_handler.cc

```diff
@@ -314,22 +315,29 @@ namespace Particles
   ParticleHandler<dim, spacedim>::particles_in_cell(
     const typename Triangulation<dim, spacedim>::active_cell_iterator &cell)
   {
-    const internal::LevelInd level_index =
-      std::make_pair(cell->level(), cell->index());
+    const unsigned int active_cell_index = cell->active_cell_index();
 
-    if (cell->is_ghost())
+    if (cell->is_artificial() == false)
       {
-        const auto particles_in_cell = ghost_particles.equal_range(level_index);
-        return boost::make_iterator_range(
-          particle_iterator(ghost_particles, particles_in_cell.first),
-          particle_iterator(ghost_particles, particles_in_cell.second));
-      }
-    else if (cell->is_locally_owned())
-      {
-        const auto particles_in_cell = particles.equal_range(level_index);
-        return boost::make_iterator_range(
-          particle_iterator(particles, particles_in_cell.first),
-          particle_iterator(particles, particles_in_cell.second));
+        std::vector<std::vector<Particle<dim, spacedim>>> &container =
+          (cell->is_locally_owned() == true) ? particles : ghost_particles;
+
+        if (container[active_cell_index].size() == 0)
+          {
+            return boost::make_iterator_range(
+              particle_iterator(container, active_cell_index, 0),
+              particle_iterator(container, active_cell_index, 0));
+          }
+        else
+          {
+            auto end =
+              particle_iterator(container,
```

Comment:
make it easier to read: if you declare an `end` variable, also declare a `begin` variable. or define both in-line as you currently do for the begin iterator